packages/kbn-openapi-bundler/src/openapi_bundler.test.ts

```diff
@@ -6,125 +6,235 @@
-    await expectBundleToMatchFile('recursive_ref_specs', 'expected.yaml');
-  });
+    it('bundles one file with a local reference', async () => {
+      const folder = join('bundle_refs', 'spec_with_local_ref');
 
-  it('bundles spec with a self-recursive reference', async () => {
-    await bundleFolder('self_recursive_ref');
-    await expectBundleToMatchFile('self_recursive_ref', 'expected.yaml');
-  });
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
 
-  it('bundles one endpoint with different versions', async () => {
-    await bundleFolder('different_endpoint_versions');
-    await expectBundleToMatchFile('different_endpoint_versions', 'expected.yaml');
-  });
+    it('bundles one file with an external reference', async () => {
+      const folder = join('bundle_refs', 'spec_with_external_ref');
 
-  it('bundles spec with different OpenAPI versions', async () => {
-    await bundleFolder('different_openapi_versions');
-    await expectBundleToMatchFile('different_openapi_versions', 'expected.yaml');
-  });
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('bundles conflicting but equal references', async () => {
+      const folder = join('bundle_refs', 'conflicting_but_equal_refs_in_different_specs');
+
+      await bundleFolder(folder);
+      await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    });
+
+    it('DOES NOT bundle external conflicting references encountered in on spec file', async () => {
+      const folder = join('bundle_refs', 'conflicting_refs');
 
-  it('bundles conflicting but equal references', async () => {
-    await bundleFolder('conflicting_but_equal_refs_in_different_specs');
-    await expectBundleToMatchFile('conflicting_but_equal_refs_in_different_specs', 'expected.yaml');
+      await expectBundlingError(folder, /\/components\/schemas\/ConflictTestSchema/);
+    });
+
+    it('DOES NOT bundle conflicting references encountered in separate specs', async () => {
+      const folder = join('bundle_refs', 'conflicting_refs_in_different_specs');
+
+      await expectBundlingError(folder, /\/components\/schemas\/ConflictTestSchema/);
+    });
   });
 
-  it('fails to bundle conflicting references encountered in separate specs', async () => {
-    await expectBundlingError(
-      'conflicting_refs_in_different_specs',
-      /\/components\/schemas\/ConflictTestSchema/
-    );
+  describe('circular references', () => {
+    // Fails because `writeYamlDocument()` has `noRefs: true` setting
+    // it('bundles recursive spec', async () => {
+    //   const folder = join('circular', 'recursive_spec');
+
+    //   await bundleFolder(folder);
+    //   await expectBundleToMatchFile(DEFAULT_BUNDLED_FILE_PATH, join(folder, 'expected.yaml'));
+    // });
```

Comment:
Let's remove the commented code and the `recursive_spec` files

Reply:
I managed to uncomment the test by allowing to write documents with circular references by modifying `writeYamlDocument`.